compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -514,9 +514,10 @@ BType resolvePatternTypeFromMatchExpr(BLangMappingMatchPattern mappingMatchPatte
         if (mappingMatchPattern.matchExpr == null) {
             return patternType;
         }
-        BType intersectionType = getTypeIntersection(
-                IntersectionContext.compilerInternalIntersectionContext(),
-                mappingMatchPattern.matchExpr.getBType(), patternType, env);
+        IntersectionContext intersectionContext = IntersectionContext.compilerInternalIntersectionContext();
+        intersectionContext.ignoreDefaultValues = true;
```

Comment:
We can probably add another method like `matchClauseIntersectionContext` that sets `ignoreDefaultValues` to true?

Reply:
Meant a different method in IntersectionContext, will send a PR to the branch.